### ISSUE TITLE
stuck on opening notebook page (7.2)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [fracte/keepnote#469]
```
Upon upgrading from 7.1 to 7.2 I was unable to open my existing note book (this 
note book has an index in an alternative location on a different drive and has 
a large number of notes).
The application sat on the opening note book page.  I allowed 40 minutes or so 
but in the end had to terminate the application through task manager.
I am running windows XP SP2 and the problem is evident with Keepnote 7.2, I 
have been able to downgrade to version 7.1 and the note book once again opens 
as normal.

```

Original issue reported on code.google.com by `scop...@gmail.com` on 17 May 2011 at 9:21
- Merged into: #482

**Comment by GoogleCodeExporter** [comment]
```
Almost same experience here.
The difference is OS;
  Ubuntu 10.04 Linux 2.6.32-31-generic x86_64 GNU/Linux

Upgraded to 0.7.2, loading notebook keeps application busy for a long time. 
Frankly speaking, I could not wait until end to see it is successfully opened 
(waited about 20-30 minutes). I killed the application. After downgrade to 
0.7.1, Application successfully opened the notebook.

Below,You can find the lines from error log. 


==============================================
KeepNote 0.7.2: Mon May 16 10:48:18 2011
enabling extension 'backup_tar'
enabling extension 'new_file'
enabling extension 'editor_insert_date'
enabling extension 'command_basics'
enabling extension 'export_html'
enabling extension 'python_prompt'
enabling extension 'keepnote'
Traceback (most recent call last):
  File "/usr/lib/python2.6/dist-packages/keepnote/gui/__init__.py", line 499, in func
    notebook.load(filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/__init__.py", line 1283, in load
    attr = self._conn.read_node(self._conn.get_rootid())
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 753, in get_rootid
    return self._read_root()["nodeid"]
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 660, in _read_root
    attr = self._read_node(None, self._filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 876, in _read_node
    attr["childrenids"] = list(self._list_children_nodeids(nodeid, path))
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 836, in <genexpr>
    return (attr["nodeid"]
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 807, in _list_children_attr
    yield self._read_node(nodeid, path2)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 876, in _read_node
    attr["childrenids"] = list(self._list_children_nodeids(nodeid, path))
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 836, in <genexpr>
    return (attr["nodeid"]
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 807, in _list_children_attr
    yield self._read_node(nodeid, path2)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 844, in _read_node
    attr = self._read_attr(metafile, self._attr_defs)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 932, in _read_attr
    self._recover_attr(filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 963, in _recover_attr
    self._move_to_lostdir(filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 556, in _move_to_lostdir
    lostdir, os.path.basename(filename),  sep=u"-")
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/__init__.py", line 104, in get_unique_filename
    assert os.path.exists(path), path
AssertionError: /home/xyz/zxc/vbn/takenote/ntbGeneral/__NOTEBOOK__/lost_found
==============================================
KeepNote 0.7.2: Mon May 16 10:59:46 2011
==============================================
KeepNote 0.7.2: Mon May 16 12:20:44 2011
enabling extension 'backup_tar'
enabling extension 'new_file'
enabling extension 'editor_insert_date'
enabling extension 'command_basics'
enabling extension 'export_html'
enabling extension 'python_prompt'
enabling extension 'keepnote'
Traceback (most recent call last):
  File "/usr/lib/python2.6/dist-packages/keepnote/gui/__init__.py", line 499, in func
    notebook.load(filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/__init__.py", line 1283, in load
    attr = self._conn.read_node(self._conn.get_rootid())
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 753, in get_rootid
    return self._read_root()["nodeid"]
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 660, in _read_root
    attr = self._read_node(None, self._filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 876, in _read_node
    attr["childrenids"] = list(self._list_children_nodeids(nodeid, path))
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 836, in <genexpr>
    return (attr["nodeid"]
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 807, in _list_children_attr
    yield self._read_node(nodeid, path2)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 876, in _read_node
    attr["childrenids"] = list(self._list_children_nodeids(nodeid, path))
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 836, in <genexpr>
    return (attr["nodeid"]
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 807, in _list_children_attr
    yield self._read_node(nodeid, path2)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 844, in _read_node
    attr = self._read_attr(metafile, self._attr_defs)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 932, in _read_attr
    self._recover_attr(filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 963, in _recover_attr
    self._move_to_lostdir(filename)
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/connection/fs/__init__.py", line 556, in _move_to_lostdir
    lostdir, os.path.basename(filename),  sep=u"-")
  File "/usr/lib/python2.6/dist-packages/keepnote/notebook/__init__.py", line 104, in get_unique_filename
    assert os.path.exists(path), path
AssertionError: /home/xyz/zxc/vbn/takenote/ntbGeneral/__NOTEBOOK__/lost_found
==============================================
KeepNote 0.7.1: Mon May 16 15:41:04 2011
enabling extension 'backup_tar'
enabling extension 'new_file'
enabling extension 'editor_insert_date'
enabling extension 'command_basics'
enabling extension 'export_html'
enabling extension 'python_prompt'
enabling extension 'keepnote'
==============================================

```

Original comment by `chn...@gmail.com` on 23 May 2011 at 8:36

**Comment by GoogleCodeExporter** [comment]
```
I am now testing a new version of keepnote 0.7.3.  If you have some free time 
you might be interested in seeing if it works for you.
http://keepnote.org/keepnote/index.shtml#download-test


```

Original comment by `matt.rasmus@gmail.com` on 13 Jun 2011 at 1:47
- Added labels: **install**

**Comment by GoogleCodeExporter** [comment]
```
Hello. 

I was running 0.7.1 for a while quite successfully - I upgraded to 0.7.2 and 
experienced same delay as others - stuck opening the notebook. 

I tried 0.7.3 and though the notebook opened, the time was quite long (relative 
to 0.7.1). What seemed nearly instant in 0.7.1 takes about 20-30 sec wait to 
open in 0.7.3 (exact same notebook)

BTW - I have been using the windows installer on a Win7 Pro machine.
```

Original comment by `ibwirel...@gmail.com` on 16 Jun 2011 at 11:33

**Comment by GoogleCodeExporter** [comment]
```
Is it just the first open though?  The first open with a new version will take 
longer for updating and consistency checks.  What about opening the notebook 
the second and third time.  Is it faster then?
```

Original comment by `matt.rasmus@gmail.com` on 17 Jun 2011 at 12:23

**Comment by GoogleCodeExporter** [comment]
```
Ok it seems the remaining issue is slow notebook opening on windows.  I'll 
merge this with the other issue.  You can read my thoughts about mtimes on 
issue 482.
```

Original comment by `matt.rasmus@gmail.com` on 21 Jun 2011 at 2:46
- Changed state: **Duplicate**

**Comment by GoogleCodeExporter** [comment]
```
issue is resolved for me on 0.7.3
```

Original comment by `scop...@gmail.com` on 30 Jun 2011 at 1:19
